tests/plugins/test_change_from_in_df.py

```diff
@@ -110,8 +115,8 @@ def test_update_base_image_inspect_broken(tmpdir, caplog, docker_tasker):
     workflow.builder.base_image = image_name
     workflow.builder.set_parent_inspection_data(image_str, dict(no_id="here"))
 
-    with pytest.raises(NoIdInspection):
-        ChangeFromPlugin(docker_tasker, workflow).run()
+    run_plugin(workflow, docker_tasker, allow_failure=True)
+    assert "raised an exception: NoIdInspection" in caplog.text
```

Comment:
when we use run_plugin it will use PreBuildPluginsRunner, which runs plugins, catches exceptions and collects result

so here all I get are logs

Reply:
Doesn't `PreBuildPluginsRunner` return the result for each plugin though?

Reply:
yes it returns results, but when plugin fails that result is None

Reply:
`PreBuildPluginsRunner` also re-raises any caught exceptions as `PluginFailedException`, doesn't it? At the end of the day, this doesn't matter that much, but I think it would work.

Reply:
change_from_in_df doesn't have is_allowed_to_fail defined, so it default to True, so it won't raise

Reply:
Ah gotcha, makes sense :+1: